main.cpp

```diff
@@ -133,256 +175,335 @@ struct CarWash
-    // brand    
+    std::string material = "Carbon Fiber"; 
     std::string brand = "Acme";
-    // need maintenance?
-    bool needmaintenance = false;
-
-    // 3 things it can do:
-    // 1) adjust seat height
-    int adjustSeatHight(int riderHeight); // returns the height of the seat
-    // 2) absorb shock from bumps
-    void shockAbsorb();
-    // 3) support weight of rider
-    void supportWeight();
+    bool needMaintenance = false;
+
+    int adjustSeatHeight(int riderHeight);
+    bool shockAbsorbFrame(bool needMaintenance);
+    bool supportWeight(bool needMaintenance);
 };
 
+int Frame::adjustSeatHeight(int riderHeight)
+{
+    return ((riderHeight / 70) + 2 ) * 3;
+}
+
+bool Frame::shockAbsorbFrame(bool needMaintenanceFrame)
+{
+    return needMaintenanceFrame ? false : true;
+}
+
+bool Frame::supportWeight(bool needMaintenanceFrame)
+{
+    return needMaintenanceFrame ? false : true;
+}
+
 struct Wheel
 {
-    // circumference
     float circumference = 30.2f;
-    // number of spokes
     int numberOfSpokes = 25;
-    // material
     std::string material = "Aluminum";
-    // brand
     std::string brand = "Acme";
-    // need maintenance?
-    bool needmaintenance = false;
-
-    // 3 things it can do:
-    // 1) spin
-    int spin(int speed); // returns spin duration 
-    // 2) absorb shock from bumps
-    void shockAbsorb();
-    // 3) support weight of rider
-    void supportWeight();
+    bool needMaintenance = false;
+
+    int spin(int speed, int distance); 
+    bool shockAbsorbWheel(bool needMaintenance);
+    bool supportWeightWheel(bool needMaintenance);
 };
 
+int Wheel::spin(int speed, int distance)
+{
+    return speed * distance;
+}
+
+bool Wheel::shockAbsorbWheel(bool needMaintenanceWheel)
+{
+    return needMaintenanceWheel ? false : true;
+}
+
+bool Wheel::supportWeightWheel(bool needMaintenanceWheel)
+{
+    // should I write it like this or how it was above? I feel like this way is less confusing with the true value being first spot in the ternary opeator after the question mark.
+    return needMaintenanceWheel == false ? true : false;
+}
+
 struct Brakes
 {
-    // type
     char type = 'B';
-    // brand
     std::string brand = "Acme";
-    // material
     std::string material = "Rubber";
-    // amount of pad left
     float padRemaining = 0.2f;
-    // need maintenance?
-    bool needmaintenance = true;
-
-    // 3 things it can do:
-    // 1) stop the bike
-    void stop();
-    // 2) slow the bike
-    void slow();
-    // 3) squeak when needing to be replaced
-    bool replaceSqueek(int padRemaining); // returns a boolean telling if the pads need replaced 
+    bool needMaintenance = true;
+
+    bool stop(bool needMaintence, float pressureApplied);
+    float slow(float pressureApplied);
+    bool replaceSqueek(float padRemaining);
 };
 
+bool Brakes::stop(bool needMaintenceBrakes, float pressureApplied)
+{
+    return (needMaintenceBrakes == false && pressureApplied > 0) ? true : false; 
+}
```

Comment:
This would be easier to understand as:
```
if( .... )
    return true;
    
return false;
```
When your ternary conditions are complex, it's best to write normal `if` expressions, so the code is easy to read and understand
- [ ] revise